lib/vdsm/storage/lvm.py

```diff
@@ -1017,49 +1017,58 @@ def getAllVgs(self):
                 self.stats.hit()
         return list(vgs.values())
 
-    def getLv(self, vgName, lvName=None):
+    def getLv(self, vgName, lvName):
         """
-        Get specific LV or all LVs in specified VG.
+        Get specific LV in specified VG.
 
         If there are any stale LVs reload the whole VG, since it would
         cost us around same efforts anyhow and these stale LVs can
         be in the vg.
 
-        We never return Stale or Unreadable LVs when
-        getting all LVs for a VG, but may return a Stale or an Unreadable
-        LV when LV name is specified as argument.
+        May return a Stale or an Unreadable LV.
 
         Arguments:
             vgName (str): VG name to query.
-            lvName (str): Optional LV name.
+            lvName (str): LV name.
 
         Returns:
-            LV nameduple if lvName is specified, otherwise list of LV
-            namedtuple for all lvs in VG vgName.
+            LV nameduple.
         """
+        # vgName, lvName
+        lv = self._lvs.get((vgName, lvName))
+        if not lv or lv.is_stale():
+            self.stats.miss()
+            # while we here reload all the LVs in the VG
+            lvs = self._reloadlvs(vgName)
+            lv = lvs.get((vgName, lvName))
```

Comment:
Here it can be `None`. In this patch I keep the code as is and is the top level `getLV` where we raise if None. In the next patch I move the raise here, closer to the getter.

Since we do not want to silence the `KeyError` but rather just raise when the key is not there, I'll follow your suggestion and change it for the brackets and catch the `KeyError` exception and raise from there.